atlasaction/action.go

```diff
@@ -119,37 +121,204 @@ func MigrateLint(ctx context.Context, client *atlasexec.Client, act *githubactio
 	})
 	url := strings.TrimSpace(resp.String())
 	act.SetOutput("report-url", url)
+	publishErr := publishResult(url, err, act)
+	if publishErr != nil {
+		act.Warningf("unable to publish lint report: %v", publishErr)
+	}
+	return err
+}
+
+func publishResult(url string, err error, act *githubactions.Action) error {
 	status := "success"
 	if err != nil {
 		status = "error"
 	}
+	ghContext, err := act.Context()
+	if err != nil {
+		return err
+	}
+	event, err := triggerEvent(ghContext)
+	if err != nil {
+		return err
+	}
 	icon := fmt.Sprintf(`<img src="https://release.ariga.io/images/assets/%v.svg"/>`, status)
+	dirName := act.GetInput("dir-name")
 	summary := fmt.Sprintf(`# Atlas Lint Report
 <div>Analyzed <strong>%v</strong> %v </div><br>
-<strong>Lint report <a href=%q>available here</a></strong>`, act.GetInput("dir-name"), icon, url)
+<strong>Lint report <a href=%q>available here</a></strong>`, dirName, icon, url)
 	act.AddStepSummary(summary)
+
+	g := GithubAPI{
+		baseURL: ghContext.APIURL,
+	}
+	prNumber := event.PullRequestNumber
+	if prNumber == 0 {
+		return fmt.Errorf("unknown pr number")
+	}
+	ghToken := act.GetInput("github-token")
+	comments, err := g.getIssueComments(prNumber, event.Repository.Name, ghToken)
+	if err != nil {
+		return err
+	}
+	r, err := generateComment(summary, dirName)
+	if err != nil {
+		return err
+	}
+	if ac := findFirst(comments, isAtlasLintCommentFor(dirName)); ac != nil {
+		err = g.updateComment(ac.Id, r, event.Repository.Name, ghToken)
+	} else {
+		err = g.createIssueComment(prNumber, r, event.Repository.Name, ghToken)
+	}
+
 	return err
 }
 
-func createContext(github *githubactions.Action) (*ContextInput, error) {
-	ghContext, err := github.Context()
+type GithubIssueComment struct {
+	Id   int    `json:"id"`
+	Body string `json:"body"`
+}
+
+type GithubAPI struct {
+	baseURL string
+}
```

Comment:
merge the type definition. Also, If the GithubAPI is public, why the fields are private?